src/document/printer.js

```diff
@@ -32,7 +32,7 @@ const MODE_FLAT = Symbol("MODE_FLAT");
 
 const CURSOR_PLACEHOLDER = Symbol("cursor");
 
-const IS_MUTABLE_FILL = Symbol("IS_MUTABLE_FILL");
+const IS_REVERSED_FILL = Symbol("IS_REVERSED_FILL");
```

Comment:
Though this works pretty well, it might be confusing. For readability, I have a couple of ideas.

Idea 1: Design a new class that can (a) perform [`.shift()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/shift) operation with O(1) and (b) produce read access to the first element with O(1) and (c) calculate current length with O(1). We can implement such class easily holding reversed array as inner field. Or if it looks still confusing, linked list can be alternative.

Idea 2: Stop mutating the parts and hold the progresses (current index) for each `fill()`  as a Map instead.